components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -0,0 +1,147 @@
+        if (inboundAuthenticationConfig != null) {
+            InboundAuthenticationRequestConfig[] inboundRequestConfigs = inboundAuthenticationConfig.
+                    getInboundAuthenticationRequestConfigs();
+            if (inboundRequestConfigs != null) {
+                for (InboundAuthenticationRequestConfig inboundRequestConfig : inboundRequestConfigs) {
+                    if (inboundRequestConfig.getInboundAuthType().equals(OAUTH2)) {
+                        Property[] props = inboundRequestConfig.getProperties();
+                        for (Property prop : props) {
+                            if (prop.getName().equalsIgnoreCase(OAUTH2_CONSUMER_SECRET)) {
+                                props = (Property[]) ArrayUtils.removeElement(props, prop);
+                                inboundRequestConfig.setProperties(props);
+                                continue;   //we are interested only on this property
+                            } else {
+                                //ignore
+                            }
+                        }
+                        continue;// we are interested only on oauth2 config. Only one will be present.
+                    } else {
+                        //ignore
+                    }
+                }
+            } else {
+
+            }
+        } else {
+            //nothing to do
+        }
+    }
+
+    private ServiceProvider addClientSecret(ServiceProvider serviceProvider, String clientType,
+                                            String tenantDomain) throws IdentityOAuthAdminException {
+
+        InboundAuthenticationConfig inboundAuthenticationConfig = serviceProvider.getInboundAuthenticationConfig();
+        if (inboundAuthenticationConfig != null) {
+            InboundAuthenticationRequestConfig[] inboundRequestConfigs = inboundAuthenticationConfig.
+                    getInboundAuthenticationRequestConfigs();
+            if (inboundRequestConfigs != null) {
+                for (InboundAuthenticationRequestConfig inboundRequestConfig : inboundRequestConfigs) {
+                    if (inboundRequestConfig.getInboundAuthType().equals(OAUTH2)) {
+                        Property[] props = inboundRequestConfig.getProperties();
+                        Property property = new Property();
+                        property.setName(OAUTH2_CONSUMER_SECRET);
+                        property.setValue(getClientSecret(inboundRequestConfig.getInboundAuthKey()));
+                        props = (Property[]) ArrayUtils.add(props, property);
+                        inboundRequestConfig.setProperties(props);
+                        continue;// we are interested only on oauth2 config. Only one will be present.
+                    } else {
+                        //ignore
+                    }
+                }
+            } else {
+
+            }
+        } else {
+            //nothing to do
+        }
```

Comment:
Will be keeping the else blocks as a best practice to avoid miss leading any latter additions to the logic.